### PR TITLE
chore: styling changes on docs

### PR DESCRIPTION
**Pull request by gabrielelpidio**
Closes #<issue>

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-08-15).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog
 
- add background transition for theme switching
- changed background color for search and theme toggle on dark and light
- removed shadow and border from search and toggle respectively
- added styling to detail tag on markdown
- move theme toggle to navbar

Also addressing #445 and #444 let me know your thoughts on the changes

---

## Screenshots

image
image

### CODE REVIEW

**Comment by juliusmarminge** [comment]
This will conflict with #447 so waiting with merge

**Comment by juliusmarminge** [comment]
I repointed the PR to `julius/docs-theming`, would you mind resolving the conflicts?

**Comment by gabrielelpidio** [comment]
Yeah, no prob I'll have a look

**Comment by gabrielelpidio** [comment]
Seems good to go @juliusmarminge

**Comment by juliusmarminge** [comment]
> Seems like its good to go @juliusmarminge

Your other PR made it conflict again :(

File: www/src/components/navigation/navbar.astro [hunk 1]
```diff
@@ -59,13 +60,17 @@ const navbarLinks: Array<{ href: string; label: string }> = [
...
-    <div>
-      <SidebarToggle currentPage={currentPage} client:idle />
-    </div>
-    <div
-      class="hidden md:block shadow-sm dark:shadow-md shadow-slate-500 dark:shadow-slate-500/10 rounded-lg w-full max-w-xs"
-    >
-      <Search client:load />
+    <div class="flex flex-grow justify-end items-center">
+      <div class={clsx("m-8 text-center", { hidden: isLanding })}>
+        <ThemeToggleButton />
```
Comment (juliusmarminge) [inline www/src/components/navigation/navbar.astro:65]: Need to load the JS here:

```suggestion
        <ThemeToggleButton client:load />
```

File: www/src/components/footer/footer.astro [hunk 2]
```diff
@@ -10,7 +10,7 @@ const { path, isBlog } = Astro.props;
 <footer
   class={path === "/"
     ? `mt-auto pb-8 pt-16 text-t3-purple-100 bg-[#131232] flex flex-col`
-    : `mt-auto pb-8 pt-16 text-slate-900 dark:text-t3-purple-100 bg-t3-purple-50 dark:bg-slate-900 flex flex-col`}
+    : `mt-auto pb-8 pt-16 text-slate-900 dark:text-t3-purple-100 bg-t3-purple-50 dark:bg-slate-900 transition-colors flex flex-col`}
```
Comment (juliusmarminge) [inline www/src/components/footer/footer.astro:13]: Use clsx here too so we dont need to duplicate the classes

**Review by juliusmarminge — CHANGES_REQUESTED** [review]

**Comment by c-ehrlich** [comment]
does it work at every screen size? (ie is there enough space in the navbar at the smallest breakpoint where everything starts getting put in the navbar, to fit the logo, docs/faq/etc labels, theme picker, and search bar? i remember when playing with it last night that it gets very tightly packed at small breakpoints.

if so lgtm once conflicts are resolved.

**Comment by juliusmarminge** [comment]
> re: theme picker in navbar:
> 
> does it work at every screen size? (ie is there enough space in the navbar at the smallest breakpoint where everything starts getting put in the navbar, to fit the logo, docs/faq/etc labels, theme picker, and search bar? i remember when playing with it last night that it gets very tightly packed at small breakpoints.
> 
> if so that change looks good to me.

https://share.cleanshot.com/KbWWqk

LGTM

File: www/src/components/footer/footer.astro (hunk 2 above)
Comment (gabrielelpidio) [reply www/src/components/footer/footer.astro]: Added change

File: www/src/components/navigation/navbar.astro (hunk 1 above)
Comment (gabrielelpidio) [reply www/src/components/navigation/navbar.astro]: Fixed thanks for letting me know :sweat_smile:

**Comment by gabrielelpidio** [comment]
Fixed conflicts, let me know if there's anything I missed

**Review by juliusmarminge — APPROVED** [review]

**Pull request merged**